drivers/gpio/gpio_pca953x.c

```diff
@@ -0,0 +1,510 @@
+	} else { /* GPIO_INT_MODE_EDGE */
+		if (trig == GPIO_INT_TRIG_BOTH) {
+			irq->falling |= BIT(pin);
+			irq->rising |= BIT(pin);
+		} else if (trig == GPIO_INT_TRIG_LOW) {
+			irq->falling |= BIT(pin);
+			irq->rising &= ~BIT(pin);
+		} else if (trig == GPIO_INT_TRIG_HIGH) {
+			irq->falling &= ~BIT(pin);
+			irq->rising |= BIT(pin);
+		}
+	}
+
+	k_sem_give(&drv_data->lock);
+
+	return 0;
+}
+
+static int gpio_pca953x_manage_callback(const struct device *dev,
+					struct gpio_callback *callback,
+					bool set)
+{
+	struct pca953x_drv_data *data = dev->data;
+
+	return gpio_manage_callback(&data->cb, callback, set);
+}
+
+/**
+ * @brief Initialization function of PCA953X
+ *
+ * This sets initial input/ output configuration and output states.
+ * The interrupt is configured if this is enabled.
+ *
+ * @param dev Device struct
+ * @return 0 if successful, failed otherwise.
+ */
+static int gpio_pca953x_init(const struct device *dev)
+{
+	const struct pca953x_config *cfg = dev->config;
+	struct pca953x_drv_data *drv_data = dev->data;
+	int rc = 0;
+
+	if (!device_is_ready(cfg->i2c_dev)) {
+		LOG_ERR("I2C device not found");
+		goto out;
+	}
+
+	/* Do an initial read, this clears the interrupt pin and sets
+	 * up the initial value of the pin state input data.
+	 */
+	rc = update_input(dev);
+	if (rc) {
+		goto out;
+	}
+
+	if (cfg->interrupt_enabled) {
+		if (!device_is_ready(cfg->gpio_int.port)) {
+			LOG_ERR("Cannot get pointer to gpio interrupt device");
+			rc = -EINVAL;
+			goto out;
+		}
+
+		drv_data->dev = dev;
+
+		k_work_init(&drv_data->work, gpio_pca953x_work_handler);
+
+		rc = gpio_pin_configure_dt(&cfg->gpio_int, GPIO_INPUT);
+		if (rc) {
+			goto out;
+		}
+
+		rc = gpio_pin_interrupt_configure_dt(&cfg->gpio_int,
+						GPIO_INT_EDGE_TO_ACTIVE);
+		if (rc) {
+			goto out;
+		}
+
+		gpio_init_callback(&drv_data->gpio_cb,
+					gpio_pca953x_init_cb,
+					BIT(cfg->gpio_int.pin));
+
+		rc = gpio_add_callback(cfg->gpio_int.port,
+					&drv_data->gpio_cb);
+	}
+out:
+	if (rc) {
+		LOG_ERR("%s init failed: %d", dev->name, rc);
+	} else {
+		LOG_INF("%s init ok", dev->name);
+	}
+	return rc;
+}
+
+static const struct gpio_driver_api api_table = {
+	.pin_configure = gpio_pca953x_config,
+	.port_get_raw = gpio_pca953x_port_read,
+	.port_set_masked_raw = gpio_pca953x_port_set_masked,
+	.port_set_bits_raw = gpio_pca953x_port_set_bits,
+	.port_clear_bits_raw = gpio_pca953x_port_clear_bits,
+	.port_toggle_bits = gpio_pca953x_port_toggle_bits,
+	.pin_interrupt_configure = gpio_pca953x_pin_interrupt_configure,
+	.manage_callback = gpio_pca953x_manage_callback,
+};
+
+#define GPIO_PCA953X_INIT(n)							\
+	static const struct pca953x_config pca953x_cfg_##n = {			\
+		.i2c_dev = DEVICE_DT_GET(DT_INST_BUS(n)),			\
+		.common = {							\
+			.port_pin_mask = GPIO_PORT_PIN_MASK_FROM_DT_INST(n),	\
+		},								\
+		.interrupt_enabled = DT_INST_NODE_HAS_PROP(n, nint_gpios),	\
+		.gpio_int = GPIO_DT_SPEC_INST_GET_OR(n, nint_gpios),		\
```

Comment:
Looks like the last argument to this macro is [missing](https://buildkite.com/zephyr/zephyr/builds/36817#7ba6692d-3e1f-4d05-a6b7-8531784e4fc5/506-668) `{0}`